File: app/src/androidTest/java/net/laurakogler/medminder/MainActivityTest.java
package net.laurakogler.medminder;

import android.app.Activity;
import android.test.ActivityInstrumentationTestCase2;
import android.widget.TextView;

public class MainActivityTest extends ActivityInstrumentationTestCase2 {
    private Activity mainActivity;

    public MainActivityTest() {
        super(MainActivity.class);
    }


    @Override
    protected void setUp() throws Exception {
        super.setUp();
        mainActivity = getActivity();
    }

    public void test_itDisplaysHelloWorld() {
        TextView textView = (TextView) mainActivity.findViewById(R.id.main_text);
        final String actual = textView.getText().toString();
        assertEquals("Hello World!", actual);
    }
}

File: app/src/androidTest/java/net/laurakogler/medminder/MainActivityTest.kt
package net.laurakogler.medminder

import android.app.Activity
import android.test.ActivityInstrumentationTestCase2
import android.widget.TextView
import junit.framework.Assert

class MainActivityTest : ActivityInstrumentationTestCase2<MainActivity>(MainActivity::class.java) {
    private var mainActivity: Activity? = null


    @Throws(Exception::class)
    override fun setUp() {
        super.setUp()
        mainActivity = activity
    }

    fun test_itDisplaysHelloWorld() {
        val textView = mainActivity!!.find<TextView>(R.id.main_text)
        val actual = textView.text.toString()
        Assert.assertEquals("Hello World!", actual)
    }
}
